state/runtime/precompiled/apex_bls_signs_verification.go

```diff
@@ -0,0 +1,112 @@
+package precompiled
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/0xPolygon/polygon-edge/bls"
+	"github.com/0xPolygon/polygon-edge/chain"
+	"github.com/0xPolygon/polygon-edge/consensus/polybft/bitmap"
+	"github.com/0xPolygon/polygon-edge/state/runtime"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/umbracle/ethgo/abi"
+)
+
+const (
+	apexBLSSingleTypeByte = 0
+)
+
+var (
+	errApexBLSSignatureVerificationInvalidInput     = errors.New("invalid input")
+	errApexBLSSignatureVerificationQuorumNotReached = errors.New("quorum not reached")
+	// apexBLSInputDataMultiABIType is the ABI signature of the precompiled contract input data
+	// (hash, signature, blsPublicKeys, bitmap)
+	apexBLSInputDataMultiABIType = abi.MustNewType("tuple(bytes32, bytes, uint256[4][], bytes)")
+	// (hash, signature, blsPublicKey)
+	apexBLSInputDataSingleABIType = abi.MustNewType("tuple(bytes32, bytes, uint256[4])")
+)
+
+// apexBLSSignatureVerification verifies the given aggregated signatures using the default BLS utils functions.
+// apexBLSSignatureVerification returns ABI encoded boolean value depends on validness of the given signatures.
+type apexBLSSignatureVerification struct {
+	domain []byte
+}
+
+// gas returns the gas required to execute the pre-compiled contract
+func (c *apexBLSSignatureVerification) gas(input []byte, _ *chain.ForksInTime) uint64 {
+	return 50000
+}
+
+// Run runs the precompiled contract with the given input.
+// Input must be ABI encoded:
+// - if first byte is 0 then other bytes are decoded as tuple(bytes32, bytes, uint256[4])
+// - otherwise other input bytes are decoded as tuple(bytes32, bytes, uint256[4][], bytes, uint8)
```

Comment:
comment doesn't match with type, uint8 is not used
apexBLSInputDataMultiABIType = abi.MustNewType("tuple(bytes32, bytes, uint256[4][], bytes)"

Reply:
https://github.com/Ethernal-Tech/blade-apex-bridge/pull/39/commits/c5f80ee1740e53b95f5d3b8c8d57a52499f44caa